pandas/util/testing.py

```diff
@@ -1560,11 +1560,11 @@ def makeBoolIndex(k=10, name=None):
 
 
 def makeIntIndex(k=10, name=None):
-    return Index(lrange(k), name=name)
```

Comment:
why are you changing this? sure this is an int index, but the ordering is not expected. Revert this, and if you really need it then make a fixture in the appropriate places

Reply:
> the ordering is not expected.

Why do _you_ expect an index of int (and not of float, str) to be sorted?

What don't you understand in the [description](https://github.com/pandas-dev/pandas/pull/19483#issue-293353518) of this PR?

Sure, I can write all the fixtures you like. It just doesn't make any sense.

Reply:
because for testing using something like arange is the expected input.

Reply:
this has nothing to do with the description. you are free to use sorted, unique, not-sorted whatever in a test, but that should be explicit. you are changing a global default.

Reply:
> because for testing using something like arange is the expected input.

If _you_ expect that, fine to me: shall I then change ``makeFloatIndex`` and ``makeStringIndex`` to return sorted indexes, so that when I shuffle them it is "explicit" that they are unsorted? I couldn't care less if standard test indexes are sorted or not, as long as they all are and I don't need to add useless lines of codes in tests.

Currently not only it's not "explicit" if tests run on sorted indexes or not - it actually depends on the type.

Reply:
In general when you run a test on different types you want to change _only_ the type. Not other characteristics of the index. Then sure, there are some unavoidable differences (e.g. a ``RangeIndex`` _is_ sorted)... but apart from that, they should be as comparable as possible, so that tests can really test the different code paths.

Reply:
> because for testing using something like arange is the expected input.

Actually, the concept itself of "expected input" (referring to _content_) is completely wrong in a test.

Reply:
pls do it my way. you are changing a function that is public facing for no real reason.

Reply:
> pls do it my way. you are changing a function that is public facing for no real reason.

Oh come on Jeff, the function is clearly not part of the API. I gave "real reasons", you didn't. I even proposed an [alternative](https://github.com/pandas-dev/pandas/pull/19483#discussion_r165623414), which you didn't even comment... you're certainly not going to convince me by asking "pls do it my way".

By the way, you are a core dev, you can always merge and then change (e.g. the new asv tests), so if you don't like my PR _and_ don't like to discuss... you don't actually need to waste your time here, just merge and fix to your liking.